idb/idb.go

```diff
@@ -106,13 +106,45 @@ func DecodeTxnRowNext(s string) (round uint64, intra uint32, err error) {
 	return
 }
 
+// OptionalUint wraps bool and uint. It has a custom marshaller below.
+type OptionalUint struct {
+	Present bool
+	Value   uint
+}
+
+// MarshalText implements TextMarshaler interface.
+func (ou OptionalUint) MarshalText() ([]byte, error) {
+	if !ou.Present {
+		return nil, nil
+	}
+	return []byte(fmt.Sprintf("%d", ou.Value)), nil
+}
+
+// UnmarshalText implements TextUnmarshaler interface.
+func (ou *OptionalUint) UnmarshalText(text []byte) error {
+	if text == nil {
+		*ou = OptionalUint{}
+	} else {
+		value, err := strconv.ParseUint(string(text), 10, 64)
+		if err != nil {
+			return err
+		}
+		*ou = OptionalUint{
+			Present: true,
+			Value:   uint(value),
+		}
+	}
+
+	return nil
+}
+
 // TxnExtra is some additional metadata needed for a transaction.
 type TxnExtra struct {
 	AssetCloseAmount uint64 `codec:"aca,omitempty"`
 	// RootIntra is set on inner transactions. Combined with the confirmation
 	// round it can be used to lookup the root transaction.
 	// The type is string to allow distinguishing between 0 and empty.
```

Comment:
Update the comment to match code changes.

Reply:
done